engine/script/template/p2pkh_test.go

```diff
@@ -0,0 +1,117 @@
+package template
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/libsv/go-bk/bec"
+	"github.com/libsv/go-bk/crypto"
+	"github.com/libsv/go-bt/bscript"
+	assert "github.com/stretchr/testify/require"
+)
+
+func TestP2PKH(t *testing.T) {
+	validTests := []struct {
+		name     string
+		satoshis uint64
+		expected []P2PKHTemplate
+	}{
+		{
+			name:     "valid input",
+			satoshis: 1000,
+			expected: []P2PKHTemplate{
+				{
+					Script:   "76a9fd88ac",
+					Satoshis: 1000,
+				},
+			},
+		},
+		{
+			name:     "zero satoshis",
+			satoshis: 0,
+			expected: []P2PKHTemplate{
+				{
+					Script:   "76a9fd88ac",
+					Satoshis: 0,
+				},
+			},
+		},
+	}
+
+	errorTests := []struct {
+		name     string
+		satoshis uint64
+	}{
+		{
+			name:     "negative satoshis",
+			satoshis: ^uint64(0), // Simulating a case that would cause an error, maximum uint64 value, bitwise NOT of 0 is -1
+		},
+	}
```

Comment:
Just a matter of taste,
but can you move it closer to the place where it it used